win32ss/user/ntuser/keyboard.c

```diff
@@ -900,6 +909,97 @@ ProcessKeyEvent(WORD wVk, WORD wScanCode, DWORD dwFlags, BOOL bInjected, DWORD d
 
     /* Get virtual key without shifts (VK_(L|R)* -> VK_*) */
     wSimpleVk = IntSimplifyVk(wVk);
+	
+	/*
+     * Handle Alt+Numpad character composition.
+     * TODO: Inserting unicode characters
+     */
+    if (bIsDown &&
+        IS_KEY_DOWN(gafAsyncKeyState, VK_MENU) &&
+        !IS_KEY_DOWN(gafAsyncKeyState, VK_CONTROL))
+    {
+        /* Check if the incoming key is a numpad digit */
+        if (wVk >= VK_NUMPAD0 && wVk <= VK_NUMPAD9)
+        {
+            UINT uDigit = wVk - VK_NUMPAD0;
+
+            if (gAltNumPadState == ALTNUM_INACTIVE)
+            {
+                gAltNumPadState = (uDigit == 0) ? ALTNUM_OEM_PREFIX : ALTNUM_ACTIVE;
+            }
+
+            if (gAltNumPadState != ALTNUM_OEM_PREFIX || gAltNumPadValue > 0)
```

Comment:
It looks like `gAltNumPadValue` always remains `0` when `gAltNumPadState == ALTNUM_OEM_PREFIX`, doesn't it?